eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/annotation/load/VariantAnnotationLineMapper.java

```diff
@@ -0,0 +1,376 @@
+
+    /**
+     * Map a line in VEP output file to {@link VariantAnnotation}
+     * @param line in VEP output
+     * @param lineNumber
+     * @return a {@link VariantAnnotation}
+     * @throws Exception
+     *
+     * Most of the code is from org.opencb.biodata.formats.annotation.io.VepFormatReader#read() with few differences:
+     *  - An empty array is initialized for Hgvs (like ConsequenceTypes);
+     *  - parseFrequencies is always true and the all line is always parsed;
+     *  - The logic to move around the file (read line) and reference to previous line (currentVariantString) are removed;
+     */
+    @Override
+    public VariantAnnotation mapLine(String line, int lineNumber) throws Exception {
+        //logger.debug("Mapping line {} to VariantAnnotation", line);
+        ConsequenceType consequenceType = new ConsequenceType();
+        String[] lineFields = line.split("\t");
+
+        Map<String,String> variantMap = parseVariant(lineFields[0], lineFields[1]);  // coordinates and alternative are only parsed once
+        VariantAnnotation currentAnnotation = new VariantAnnotation(variantMap.get("chromosome"),
+                Integer.valueOf(variantMap.get("start")),
+                Integer.valueOf(variantMap.get("end")), variantMap.get("reference"),
+                variantMap.get("alternative"));
+
+        // Initialize list of consequence types
+        if(currentAnnotation.getConsequenceTypes()==null) {
+            currentAnnotation.setConsequenceTypes(new ArrayList<>());
+        }
+
+        // Initialize Hgvs
+        if(currentAnnotation.getHgvs()==null) {
+            currentAnnotation.setHgvs(new ArrayList<>());
+        }
+
+        /**
+         * parses extra column and populates fields as required. Some lines do not have extra field and end with a \t: the split function above does not return that field
+         * true parameter indicates the function to also parse frequencies
+         */
+        if(lineFields.length>13) {
+            parseExtraField(consequenceType, lineFields[13], currentAnnotation);
+        }
+
+        // Remaining fields only of interest if the feature is a transcript
+        if(lineFields[5].toLowerCase().equals("transcript")) {
+            parseRemainingFields(consequenceType, lineFields);
+            // Otherwise just set SO terms
+        } else {
+            consequenceType.setSoTermsFromSoNames(Arrays.asList(lineFields[6].split(",")));   // fill so terms
+        }
+        currentAnnotation.getConsequenceTypes().add(consequenceType);
+        
+        return currentAnnotation;
+    }
+
+    /**
+     * From org.opencb.biodata.formats.annotation.io.VepFormatReader
+     * #parseRemainingFields(org.opencb.biodata.models.variant.annotation.ConsequenceType, java.lang.String[])
+     */
+    private void parseRemainingFields(ConsequenceType consequenceType, String[] lineFields) {
```

Comment:
This method could be renamed to `parseTranscriptFields()` for clearer description of its task. Field assignment doesn't follow order found (6th index is the last to assign).